sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -911,61 +909,53 @@ struct include_files
   }
   std::vector<std::pair<std::string, std::string>> record;
 };
-using include_files_key = include_files;
-
-template <>
-struct is_property_key_of<include_files_key,
-                          detail::create_bundle_from_source_props>
-    : std::true_type {};
 
 /////////////////////////
 // PropertyT syclex::build_options
 /////////////////////////
-struct build_options
-    : detail::run_time_property_key<detail::PropKind::BuildOptions> {
+struct build_options : new_properties::detail::property_base<build_options> {
+  static constexpr std::string_view property_name{
+      "sycl::ext::oneapi::experimental::build_options"};
   std::vector<std::string> opts;
   build_options(const std::string &optsArg) : opts{optsArg} {}
   build_options(const std::vector<std::string> &optsArg) : opts(optsArg) {}
 };
-using build_options_key = build_options;
-
-template <>
-struct is_property_key_of<build_options_key, detail::build_source_bundle_props>
-    : std::true_type {};
 
 /////////////////////////
 // PropertyT syclex::save_log
 /////////////////////////
-struct save_log : detail::run_time_property_key<detail::PropKind::BuildLog> {
+struct save_log : new_properties::detail::property_base<save_log> {
+  static constexpr std::string_view property_name{
+      "sycl::ext::oneapi::experimental::save_log"};
   std::string *log;
   save_log(std::string *logArg) : log(logArg) {}
 };
-using save_log_key = save_log;
-
-template <>
-struct is_property_key_of<save_log_key, detail::build_source_bundle_props>
-    : std::true_type {};
 
 /////////////////////////
 // PropertyT syclex::registered_kernel_names
 /////////////////////////
 struct registered_kernel_names
-    : detail::run_time_property_key<detail::PropKind::RegisteredKernelNames> {
+    : new_properties::detail::property_base<registered_kernel_names> {
+  static constexpr std::string_view property_name{
+      "sycl::ext::oneapi::experimental::registered_kernel_names"};
   std::vector<std::string> kernel_names;
   registered_kernel_names() {}
   registered_kernel_names(const std::string &knArg) : kernel_names{knArg} {}
   registered_kernel_names(const std::vector<std::string> &knsArg)
       : kernel_names(knsArg) {}
   void add(const std::string &name) { kernel_names.push_back(name); }
 };
-using registered_kernel_names_key = registered_kernel_names;
-
-template <>
-struct is_property_key_of<registered_kernel_names_key,
-                          detail::build_source_bundle_props> : std::true_type {
-};
 
 namespace detail {
+template <typename property_list_ty>
+inline constexpr bool are_properties_valid_for_create_bundle_from_source =
+    new_properties::all_properties_in_v<property_list_ty, include_files>;
+
+template <typename property_list_ty>
+inline constexpr bool are_properties_valid_for_build_source_bundle =
+    new_properties::all_properties_in_v<property_list_ty, build_options,
+                                        save_log, registered_kernel_names>;
+
```

Comment:
This is subjective, but I think this is better than having each property specialize `is_property_key_of` as all the supported properties of something are listed in one place.